presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -296,6 +314,14 @@ protected Type visitCurrentTime(CurrentTime node, StackableAstVisitorContext<Con
         @Override
         protected Type visitSymbolReference(SymbolReference node, StackableAstVisitorContext<Context> context)
         {
+            if (context.getContext().isInLambda()) {
```

Comment:
`StackableAstVisitorContext` was introduced in order to provide an information about placement of currently processed node in AST. I think it has a value to extend to support such generic problems to know if this node is wrapped in another node (even if there are nodes between). Thanks to that it could be then used in other such cases as well.

Nevertheless, I understand that you will need your own context anyway.